src/libpriv/rpmostree-kargs-process.c

```diff
@@ -0,0 +1,624 @@
+ *  this as a replacement.
+ *
+ *  Returns: False if can not find the string value in the array
+ *
+ **/
+static
+gboolean _ostree_ptr_array_find (GPtrArray       *array,
+                                 const char      *val,
+                                 int             *out_index)
+{
+  for (int counter = 0; counter < array->len; counter++)
+    {
+      const char *temp_val = array->pdata[counter];
+      if  (g_str_equal (val, temp_val))
+        {
+          *out_index = counter;
+          return TRUE;
+        }
+    }
+    *out_index = 0; /* default to zero if not found */
+    return FALSE;
+}
+
+/*
+ * Note: this function is newly added to the API
+ *
+ * _ostree_query_arg_status:
+ * @kargs: A OstreeKernelArg instance
+ * @arg: a string to 'query status' (find its validity)
+ * @out_query_flag: a OstreeKernelArgQueryFlag, used to tell the caller result of finding
+ * @is_replaced: tell if the caller is from replace or delete
+ * @out_index: if successfully found, return the arg's index
+ * @error: an error instance
+ *
+ * This function provides check for the argument string
+ * to see if it is valid for deletion/replacement. More
+ * detailed explanation can be found in delete/replace section.
+ *
+ * Returns: False if an error is set
+ */
+static gboolean
+_ostree_kernel_arg_query_status (OstreeKernelArgs         *kargs,
+                                 const char               *arg,
+                                 OstreeKernelArgQueryFlag *out_query_flag,
+                                 gboolean                  is_replaced,
+                                 int                      *out_index,
+                                 GError                   **error)
+{
+  g_autofree char *duped = g_strdup (arg);
+  g_autofree char *val = g_strdup (split_keyeq (duped));
+
+  /* For replaced, it is a special case, we allow
+   * key=value=new_value, thus, this split is to
+   * disgard the new value if there is one */
+  const char *replaced_val = split_keyeq (val);
+
+  GPtrArray *values = g_hash_table_lookup (kargs->table, duped);
+
+  if (!values)
+    {
+      g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                           "The key is not found, type rpm-ostree ex kargs to see what args are there");
+      return FALSE;
+    }
+
+  /* See if the value is inside the value list */
+  if (!_ostree_ptr_array_find (values, val, out_index))
+    {
+      /* Both replace and delete support empty value handlation
+       * thus adding a condition here to handle it separately */
+      if (!*val && values->len ==1)
+        {
+          *out_query_flag = OSTREE_KERNEL_ARG_KEY_ONE_VALUE;
+          return TRUE;
+        }
+      /* This is a special case for replacement where
+       * there is only one single key, and the second val
+       * will now represent the new value (no second split
+       * will happen this case) */
+      else if (val && is_replaced && values->len == 1)
```

Comment:
`const gboolean single_value = values->len == 1` above too?

Hm...and does this need to be `if (*val && ...`?  (With the extra `*`?)  I

Reply:
> Hm...and does this need to be if (*val && ...? (With the extra *?)

yup, thanks for catching it. Did not notice it when I checked it.